test/integration/main_test.go

```diff
@@ -348,19 +357,19 @@ func testGetFirstPmapEventWithRetries(ctx context.Context, tb testing.TB, bqQuer
 
 		// Early exit retry if queried pmap event already found.
 		if len(results) > 0 {
-			pmapEvents = results
+			logEntries = results
 			return nil
 		}
 		tb.Log("Matching entry not found, retrying...")
 		return retry.RetryableError(fmt.Errorf("no matching pmap event found in bigquery after timeout"))
 	}); err != nil {
 		tb.Fatalf("Retry failed: %v.", err)
 	}
-	return pmapEvents[0]
+	return logEntries[0].PmapEvent, logEntries[0].ProcessErr
 }
 
 // queryPmapEvents queries the BQ and checks if pmap events matched the query exists or not and return the results.
-func queryPmapEvents(ctx context.Context, query *bigquery.Query) ([]*v1alpha1.PmapEvent, error) {
+func queryPmapEvents(ctx context.Context, query *bigquery.Query) ([]*QueryResult, error) {
```

Comment:
The change of such function looks weird to me. I can't understand it. Can you explain?

Reply:
new changed to `queryBQEntries` and returns BQ entires.